all.jsonnet

```diff
@@ -33,36 +33,60 @@ local b = t.bucket + commonConfig + {
   },
 };
 
-local c = t.compact + t.compact.withVolumeClaimTemplate + t.compact.withServiceMonitor + commonConfig + {
+local c = t.compact +
+          t.compact.withVolumeClaimTemplate +
+          t.compact.withServiceMonitor +
+          commonConfig + {
   config+:: {
     name: 'thanos-compact',
     replicas: 1,
   },
 };
 
-local re = t.receive + t.receive.withVolumeClaimTemplate + t.receive.withServiceMonitor + commonConfig + {
+local re = t.receive + t.receive.withVolumeClaimTemplate +
+           t.receive.withServiceMonitor +
+           commonConfig + {
   config+:: {
     name: 'thanos-receive',
     replicas: 1,
     replicationFactor: 1,
   },
 };
 
-local ru = t.rule + t.rule.withVolumeClaimTemplate + t.rule.withServiceMonitor + commonConfig + {
+local ru = t.rule + t.rule.withVolumeClaimTemplate +
+           t.rule.withServiceMonitor +
+           commonConfig + {
   config+:: {
     name: 'thanos-rule',
     replicas: 1,
   },
 };
 
-local s = t.store + t.store.withVolumeClaimTemplate + t.store.withServiceMonitor + commonConfig + {
+local s = t.store + t.store.withVolumeClaimTemplate +
+          t.store.withServiceMonitor +
+          commonConfig + {
   config+:: {
     name: 'thanos-store',
     replicas: 1,
   },
 };
 
-local q = t.query + t.query.withServiceMonitor + commonConfig + {
+local swm = t.store +
+            t.store.withVolumeClaimTemplate +
+            t.store.withServiceMonitor + t.store.withMemcachedIndexCache +
+            commonConfig + {
+  config+:: {
+    name: 'thanos-store',
+    replicas: 1,
+    memcached+: {
+      addresses: [],
```

Comment:
can at least add comment how to add your addresses?

Or can we add some k8s memcached service ? Example one or maybe it should be part of mixin (memcached statefulset)